src/main/java/org/fenixedu/academicextensions/services/registrationhistory/RegistrationHistoryReportService.java

```diff
@@ -312,6 +318,13 @@ private boolean isValidGraduated(RegistrationHistoryReport report, ProgramConclu
             return false;
         }
 
+        String number = conclusionBean.getConclusionNumber();
+        if (programConclusionNumber != null && StringUtils.isNotBlank(programConclusionNumber)) {
```

Comment:
Could we simply this test using ?
`StringUtils.isBlank(programConclusionNumber) || Objects.equals(conclusionBean.getConclusionNumber(), programConclusionNumber)`